atomic_reactor/plugins/pre_bump_release.py

```diff
@@ -8,315 +8,64 @@
     def run(self):
         """
         run the plugin
         """
 
-        if self.workflow.build_process_failed:
-            self.log.info("Build already failed, not incrementing release")
+        parser = DockerfileParser(self.workflow.builder.df_path)
+        release_labels = get_all_label_keys('release')
+        dockerfile_labels = parser.labels
+        if any(release_label in dockerfile_labels
+               for release_label in release_labels):
+            self.log.debug("release set explicitly so not incrementing")
             return
 
-        # Ensure we can use the git repository already checked out for us
-        source = self.workflow.source
-        assert isinstance(source, GitSource)
-        with GitRepo(source.get(), self.log) as repo:
-            # Note: when this plugin is configured by osbs-client,
-            # source.git_commit (the Build's source git ref) comes from
-            # --git-branch not --git-commit. The value from --git-commit
-            # went into our self.git_ref.
-            branch = source.git_commit
-            try:
-                branch_sha = repo.git(['rev-parse', branch])
-            except subprocess.CalledProcessError:
-                self.log.error("Branch '%s' not found in git repo",
-                               source.git_commit)
-                raise RuntimeError("Branch '%s' not found" % branch)
-
-            # We checked out the right branch
-            assert repo.git(['rev-parse', 'HEAD']) == branch_sha
-
-            # We haven't reset it to an earlier commit
-            remote = repo.git(['config', '--get',
-                               'branch.{branch}.remote'.format(branch=branch)])
-            upstream = '{remote}/{branch}'.format(remote=remote, branch=branch)
-            upstream_sha = repo.git(['rev-parse', upstream])
-            assert branch_sha == upstream_sha
+        # No release labels are set so set them.
+        component_label = get_preferred_label_key(dockerfile_labels,
+                                                  'com.redhat.component')
+        try:
+            component = dockerfile_labels[component_label]
+        except KeyError:
+            raise RuntimeError("missing label: {}".format(component_label))
 
-            if is_rebuild(self.workflow):
-                self.log.info("Incrementing release label")
-                self.bump(repo, remote)
-            else:
-                self.log.info("Verifying branch is at specified commit")
-                self.verify_branch(branch, branch_sha)
+        latest = self.xmlrpc.getLatestBuilds(self.target, package=component)
+        try:
+            next_release = self.xmlrpc.getNextRelease(latest[0])
+        except IndexError:
+            next_release = 1
+
+        next_release = str(next_release)
+        for release_label in release_labels:
+            self.log.info("setting %s=%s", release_label, next_release)
+            dockerfile_labels[release_label] = next_release
```

Comment:
`dockerfile_labels` is a `Labels` instance at this point. I should make that clearer.